packages/shared/src/api_types.ts

```diff
@@ -18,11 +18,16 @@ export interface GameResponse {
   time_control?: ITimeControlBase;
 }
 
+// We may add more roles like "moderator" or "bot" in the future
+export type UserRole = "admin";
+
 export interface UserResponse {
-  id?: string;
+  id: string;
   login_type: "guest" | "persistent";
   username?: string;
   rating?: number;
+  // undefined is just a normal user.
+  role?: UserRole;
```

Comment:
Is `UserResponse` an extension of `User`?

Reply:
Basically. I avoided it in this PR, but I think we should just delete `User` now that these types are compatible.

My assumption is that `User` just existed because of the ID nullability issues (addressed in this PR)